File: Examples/DeveloperObjcExample/ObjcExample/EventsViewController.m
//
//  EventsViewController.m
//  AppcuesObjcExample
//
//  Created by Matt on 2022-06-01.
//

#import "EventsViewController.h"
#import "Appcues+Shared.h"

@interface EventsViewController ()

@end

@implementation EventsViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}

- (void)viewDidAppear:(BOOL)animated {
    [super viewDidAppear:animated];

    [[Appcues shared] screenWithTitle:@"Trigger Events" properties:nil];
}

- (IBAction)buttonOneTapped:(UIButton *)sender {
    [[Appcues shared] trackWithName:@"event1" properties:nil];
}

- (IBAction)buttonTwoTapped:(UIButton *)sender {
    [[Appcues shared] trackWithName:@"event2" properties:nil];
}

- (IBAction)debugTapped:(UIBarButtonItem *)sender {
    [[Appcues shared] debug];
}

@end

File: Examples/DeveloperSPMExample/SPMExample/EventsViewController.swift
//
//  EventsViewController.swift
//  AppcuesExample
//
//  Created by Matt on 2021-10-12.
//

import UIKit
import AppcuesKit

class EventsViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
    }

    override func viewDidAppear(_ animated: Bool) {
        super.viewDidAppear(animated)

        Appcues.shared.screen(title: "Trigger Events")
    }

    @IBAction private func buttonOneTapped(_ sender: UIButton) {
        Appcues.shared.track(name: "event1")
    }

    @IBAction private func buttonTwoTapped(_ sender: UIButton) {
        Appcues.shared.track(name: "event2")
    }

    @IBAction private func debugTapped(_ sender: Any) {
        Appcues.shared.debug()
    }
}
